packages/wallet/backend/src/paymentPointer/webMonetization/service.ts

```diff
@@ -0,0 +1,148 @@
+import { AccountService } from '@/account/service'
+import { Env } from '@/config/env'
+import { Conflict, NotFound } from '@/errors'
+import { RafikiClient } from '@/rafiki/rafiki-client'
+import { CacheService } from '../../cache/service'
+import { PaymentPointer } from '../model'
+import { UpdatePaymentPointerArgs } from '../service'
+import { WMPaymentPointer } from './model'
+
+
+
+type UpdateWMPaymentPointerArgs = UpdatePaymentPointerArgs & {
+    assetCode: string
+    assetScale: number
+    balance: number
+  }
+
+interface IWMPaymentPointerService {
+  create: (params: CreateWMPaymentPointerParams) => Promise<PaymentPointer>
+  getById: (
+    userId: string,
+    accountId: string,
+    id: string
+  ) => Promise<PaymentPointer>
+  updateBalance: (args: UpdateWMPaymentPointerArgs) => Promise<void>
+}
+
+interface WMPaymentPointerServiceDependencies {
+  accountService: AccountService
+  rafikiClient: RafikiClient
+  env: Env,
+  cache: CacheService
+}
+
+
+type CreateWMPaymentPointerParams = {
+    userId: string,
+    accountId: string,
+    paymentPointerName: string,
+    publicName: string,
+    assetScale: number,
+    assetCode: string,
+}
+
+
+export class WMPaymentPointerService implements IWMPaymentPointerService {
+  constructor(private deps: WMPaymentPointerServiceDependencies) {}
+
+
+  async create(
+   params: CreateWMPaymentPointerParams
+  ): Promise<PaymentPointer> {
+    const account = await this.deps.accountService.findAccountById(
+        params.accountId,
+        params.userId
+    )
+
+    const url = `${this.deps.env.OPEN_PAYMENTS_HOST}/${params.paymentPointerName}`
+    let wmPaymentPointer = await WMPaymentPointer.query().findOne({ url })
+
+    if (wmPaymentPointer) {
+      if (wmPaymentPointer.accountId != params.accountId || account.userId !== params.userId) {
+        throw new Conflict(
+          'This payment pointer already exists. Please choose another name.'
+        )
+      }
+    }
+
+    const rafikiPaymentPointer = await this.deps.rafikiClient.createRafikiPaymentPointer(
+        params.publicName,
+        account.assetId,
+        url
+    )
+
+      wmPaymentPointer = await WMPaymentPointer.query().insert({
+        assetCode: params.assetCode,
+        assetScale: params.assetScale,
+        url: rafikiPaymentPointer.url,
+        publicName: params.publicName,
+        accountId: params.accountId,
+        id: rafikiPaymentPointer.id
+      })
+
+    await this.deps.cache.set(wmPaymentPointer.id, wmPaymentPointer, {expiry: 60});
+    
+    return wmPaymentPointer
+  }
+
+  async getById(
+    userId: string,
+    accountId: string,
+    id: string
+  ): Promise<WMPaymentPointer> {
+    const cacheHit = await this.deps.cache.get<WMPaymentPointer>(id);
```

Comment:
the method that will be used in webhooks should check the cache and request the db based on id only (with active: true check) - the check for account and user is not needed for this case

Reply:
Thanks. I had already moved it out, have another look 🙏